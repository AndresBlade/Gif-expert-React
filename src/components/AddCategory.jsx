import PropTypes from 'prop-types'
import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value)
    }

    const onSubmit = e => {
        e.preventDefault()
        if (inputValue.trim().length <= 1) return
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

    return (
        <form name='gif-search' onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}