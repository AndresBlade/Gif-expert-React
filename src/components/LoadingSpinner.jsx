import React from 'react'
import '../spinner.css'

export const LoadingSpinner = ({ isLoading }) => {
    return (
        isLoading && (<div className="spinner-container" data-testid='spinner'>
            <div className="spinner"></div>
        </div>)
    )
}
