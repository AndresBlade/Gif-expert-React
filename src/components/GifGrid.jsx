import PropTypes from "prop-types"
import { GifItem } from "./GifItem"
import { useFetchGifs } from '../hooks/useFetchGifs'
import { LoadingSpinner } from "./LoadingSpinner"

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category)

    // const [images, setImages] = useState([])

    // const getImages = async () => {
    //     const newImages = await getGifs(category)
    //     setImages(newImages)
    // }

    // useEffect(() => {
    //     getImages()
    // }, [])

    return (
        <>
            <LoadingSpinner isLoading={isLoading} />
            <div className="card-grid">
                {
                    images.map((image) =>
                        <GifItem key={image.id} {...image} />
                    )}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


