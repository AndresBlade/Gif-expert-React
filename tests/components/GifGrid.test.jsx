import { render, screen } from '@testing-library/react'
import { GifGrid } from '../../src/components/GifGrid'
import { useFetchGifs } from '../../src/hooks/useFetchGifs'

jest.mock('../../src/hooks/useFetchGifs')

describe('Pruebas en <GifGrid />', () => {

    const category = 'Peluches'

    test('Debería de hacer match con el snapshot', async () => {
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });
        const { container } = render(<GifGrid category={category} />)
        expect(container).toMatchSnapshot()
    })

    test('Debería de mostrar el loading inicialmente', async () => {
        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        });
        render(<GifGrid category={category} />)
        expect(screen.getByTestId('spinner')).toBeInTheDocument()
    })

    test('Debe de mostrar items cuando se cargan las imagenes UseFetchGifs', async () => {

        const gifs = [{
            id: 'ABC',
            title: 'Peluche Viejo',
            url: 'https://peluches.com'
        },
        {
            id: '123',
            title: 'Peluches nuevos en oferta',
            url: 'https://peluchesOferta.com'
        }]

        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: false,
        });

        render(<GifGrid category={category} />)

        screen.debug()

        expect(screen.getAllByRole('img')).toHaveLength(2)

    })
})