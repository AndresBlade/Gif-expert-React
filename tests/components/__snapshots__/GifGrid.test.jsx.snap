// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Deber√≠a de hacer match con el snapshot 1`] = `
<div>
  <div
    class="spinner-container"
    data-testid="spinner"
  >
    <div
      class="spinner"
    />
  </div>
  <div
    class="card-grid"
  />
</div>
`;
